src/mon/Monitor.cc

```diff
@@ -5961,9 +5995,28 @@ void Monitor::tick()
   }
 
   mgr_client.update_daemon_health(get_health_metrics());
+  check_memory_usage();
   new_tick();
 }
 
+void Monitor::check_memory_usage()
+{
+  static MemoryModel mm(g_ceph_context);
+  static MemoryModel::snap last;
+  mm.sample(&last);
+  static MemoryModel::snap baseline = last;
```

Comment:
These should never have been static variables in the MDS. Please move it to `Monitor` (and fix osd/mds as well).